File: meetcode.client/src/types/request/problem/problemAddRequest.ts
import { Difficulty } from "../../admin/problem";
import { ProblemTag } from "../../admin/problemTag";
import { ProblemTemplateAddRequest } from "../problemTemplate/problemTemplateAddRequest";

export type TagOption = {
    tagId: string;
    name: string;
}

export interface ProblemAddRequest {
    title: string;
    difficulty: Difficulty;
    statementMd: string;
    timeLimitMs: number;
    memoryLimitMb: number;
    createdBy: string;
    tagIds: string[];
}

File: MeetCode.Application/DTOs/Request/Problem/ProblemAddRequest.cs

using MeetCode.Application.DTOs.Request.ProblemTemplate;

namespace MeetCode.Application.DTOs.Request.Problem
{
    public sealed record ProblemAddRequest(
        string Title, 
        string StatementMd,
        string Difficulty,
        int TimeLimitMs,
        int MemoryLimitMb,
        List<Guid> TagIds
        );

}
